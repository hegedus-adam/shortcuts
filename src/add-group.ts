import { IGroup } from './data';
import { addLink } from './add-link';

export function addGroup(group: IGroup) {
    const section = document.createElement('section');
    const h2 = document.createElement('h2');
    h2.innerHTML = group.groupLabel;
  
    const ul = document.createElement('ul');
    group.links.forEach(link => {
      ul.appendChild(addLink(link));
    });
  
    section.appendChild(h2);
    section.appendChild(ul);
  
    return section;
  }
