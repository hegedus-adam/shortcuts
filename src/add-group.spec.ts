import { addGroup } from './add-group';
import { JSDOM } from 'jsdom';

const dom = new JSDOM();
global.document = dom.window.document;
global.window = dom.window;

describe('addGroup', () => {
  it('should append link', () => {
    const group = {
      groupLabel: 'Group Name',
      groupId: 0,
      links: [
        {
          id: 0,
          label: 'Test Label',
          url: 'https://somewhere.to',
        },
      ],
    };

    const actual = addGroup(group);

    expect(actual).toMatchSnapshot();
  });
});
