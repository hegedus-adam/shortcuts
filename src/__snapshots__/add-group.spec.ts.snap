// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addGroup should append link 1`] = `
<section>
  <h2>
    Group Name
  </h2>
  <ul>
    <li>
      <a
        class="link"
        href="https://somewhere.to"
      >
        Test Label
      </a>
    </li>
  </ul>
</section>
`;
